src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Disposables/Disposables/DisposableStruct.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+
+namespace Uno.Disposables;
+
+/// <summary>
+/// This Disposable is used with using statements to imitate golang-style defer statements and RAII-like logic
+/// without allocating or boxing.
+/// </summary>
+/// <remarks>
+/// Make sure that the <paramref name="disposingAction"/> doesn't capture anything (or even better, is static)
+/// to avoid allocations. If you need multiple parameters for the <paramref name="disposingAction"/>, you can
+/// bundle them in a tuple.
+/// </remarks>
+/// <param name="disposingAction">The action that is fired on disposing.</param>
+/// <param name="argToAction">The argument supplied to the <paramref name="disposingAction"/></param>
+public struct DisposableStruct<T>(Action<T> disposingAction, T argToAction) : IDisposable
+{
+	private bool _disposed;
+
+	public void Dispose()
+	{
+		if (_disposed)
+		{
+			throw new InvalidOperationException($"Disposing {nameof(DisposableStruct<T>)} twice");
+		}
+		_disposed = true;
+		disposingAction(argToAction);
+	}
+
+	public bool Equals(DisposableStruct<T> other) => _disposed == other._disposed;
```

Comment:
`disposingAction` should be considered in equality. I'm not sure though why do we need to implement equality though

Reply:
We don't, I just added these because I was getting CA1815 errrors.

Reply:
It is usually fine to suppress the warning in this case. But if we will implement equality, it has to be correct.